### ISSUE TITLE
UI doesn't update on window expand

### ISSUE THREAD

**Issue by tbillington** [bevyengine/bevy#8124]
## Bevy version

0.10

## What you did

Wrapping flex items aren't reflowing/recalculating when expanding the window after shrinking it.

https://user-images.githubusercontent.com/2771466/226109988-5531b9f5-75d9-42ac-81c2-9217ef3a4250.mov

```rust
commands
    .spawn(NodeBundle {
        style: Style {
            size: Size::all(Val::Percent(100.)),
            align_items: AlignItems::FlexStart,
            ..Default::default()
        },
        ..Default::default()
    })
    .with_children(|p| {
        p.spawn(NodeBundle {
            style: Style {
                flex_wrap: FlexWrap::Wrap,
                gap: Size::width(Val::Px(8.)),
                ..Default::default()
            },
            ..Default::default()
        })
        .with_children(|p| {
            for color in COLOR_PALETTE {
                p.spawn(NodeBundle {
                    style: Style {
                        size: Size::all(Val::Px(32.)),
                        ..Default::default()
                    },
                    background_color: BackgroundColor(Color::WHITE),
                    ..Default::default()
                })
```

**Comment by ickshonpe** [comment]
This is weird, it looks like #8111 but that only happens with the custom measure func.

I tried your example code and it seemed to work fine
```
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(setup)
        .run();
}

fn setup(mut commands: Commands) {
    commands.spawn(Camera2dBundle::default());
    commands
        .spawn(NodeBundle {
            style: Style {
                size: Size::all(Val::Percent(100.)),
                align_items: AlignItems::FlexStart,
                ..Default::default()
            },
            ..Default::default()
        })
        .with_children(|p| {
            p.spawn(NodeBundle {
                style: Style {
                    flex_basis: Val::Percent(100.),
                    flex_wrap: FlexWrap::Wrap,
                    gap: Size::width(Val::Px(8.)),
                    ..Default::default()
                },
                ..Default::default()
            })
            .with_children(|p| {
                for color in [
                    Color::RED,
                    Color::GREEN,
                    Color::BLUE,
                    Color::WHITE,
                    Color::AQUAMARINE,
                    Color::YELLOW,
                    Color::RED,
                    Color::GREEN,
                    Color::BLUE,
                    Color::WHITE,
                    Color::AQUAMARINE,
                    Color::YELLOW,
                    Color::RED,
                    Color::GREEN,
                    Color::BLUE,
                    Color::WHITE,
                    Color::AQUAMARINE,
                    Color::YELLOW,
                    Color::RED,
                    Color::GREEN,
                    Color::BLUE,
                    Color::WHITE,
                    Color::AQUAMARINE,
                    Color::YELLOW,
                ] {
                    p.spawn(NodeBundle {
                        style: Style {
                            size: Size::all(Val::Px(32.)),
                            ..Default::default()
                        },
                        background_color: BackgroundColor(color),
                        ..Default::default()
                    });
                }
            });
        });
}

```

But then I added `flex_direction: FlexDirection::Column` to the outer node and it's completely broken, much worse than in your demo clip.

**Comment by ickshonpe** [comment]
Looks like it's related to #8111

**Comment by nicoburns** [comment]
@tbillington This should be fixed by Taffy `v0.3.9`. You can obtain the new version of Taffy by running `cargo update`
